File: backend/src/main/java/com/example/scmbackend/purchaseorder/PurchaseOrder.java
package com.example.scmbackend.purchaseorder;

import com.example.scmbackend.supplier.Supplier;
import com.example.scmbackend.warehouse.Warehouse;
import jakarta.persistence.*;
import lombok.Getter;
import lombok.Setter;
import lombok.NoArgsConstructor;
import lombok.AllArgsConstructor;

import java.time.LocalDate;
import java.util.List;

@Entity
@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
public class PurchaseOrder {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @ManyToOne
    @JoinColumn(name = "supplier_id")
    private Supplier supplier;

    @ManyToOne
    @JoinColumn(name = "warehouse_id")
    private Warehouse warehouse;

    private LocalDate orderDate;

    private String status; // e.g. "PENDING", "RECEIVED"

    @OneToMany(mappedBy = "purchaseOrder", cascade = CascadeType.ALL, orphanRemoval = true)
    private List<PurchaseOrderItem> items;
}

File: frontend/src/types/purchaseorder.ts
export interface PurchaseOrderItem {
  id: number;
  productId: number;
  productName: string;
  quantity: number;
}

export interface PurchaseOrder {
  id: number;
  supplierId: number;
  supplierName: string;
  warehouseId: number;
  warehouseName: string;
  orderDate: string;
  status: string;
  items: PurchaseOrderItem[];
}